tasks/quality_gates.py

```diff
@@ -0,0 +1,144 @@
+import os
+import traceback
+import typing
+
+import yaml
+from invoke import task
+
+from tasks.github_tasks import pr_commenter
+from tasks.libs.ciproviders.github_api import GithubAPI
+from tasks.libs.common.color import color_message
+from tasks.static_quality_gates.lib.gates_lib import GateMetricHandler
+
+FAIL_CHAR = "❌"
+SUCCESS_CHAR = "✅"
+
+body_pattern = """### {}
+
+|Result|Quality gate|On disk size|On disk size limit|On wire size|On wire size limit|
+|----|----|----|----|----|----|
+"""
+
+body_error_footer_pattern = """<details>
+<summary>Gate failure full details</summary>
+
+|Quality gate|Error type|Error message|
+|----|---|--------|
+"""
+
+
+def display_pr_comment(
+    ctx, final_state: bool, gate_states: list[dict[str, typing.Any]], metric_handler: GateMetricHandler
+):
+    """
+    Display a comment on a PR with results from our static quality gates checks
+    :param ctx: Invoke task context
+    :param final_state: Boolean that represents the overall state of quality gates checks
+    :param gate_states: State of each quality gate
+    :param metric_handler: Precise metrics of each quality gate
+    :return:
+    """
+    title = f"Static quality checks {SUCCESS_CHAR if final_state else FAIL_CHAR}"
+    body_info = body_pattern.format("Info")
+    body_error = body_pattern.format("Error")
+    body_error_footer = body_error_footer_pattern
+
+    with_error = False
+    with_info = False
+    # Sort gates by error_types to group in between NoError, AssertionError and StackTrace
+    for gate in sorted(gate_states, key=lambda x: x["error_type"] is None):
+
+        def getMetric(metric_name, gate_name=gate['name']):
+            try:
+                return metric_handler.get_formated_metric(gate_name, metric_name)
+            except KeyError:
+                return "DataNotFound"
+
+        if gate["error_type"] is None:
+            body_info += f"|{SUCCESS_CHAR}|{gate['name']}|{getMetric('current_on_disk_size')}|{getMetric('max_on_disk_size')}|{getMetric('current_on_wire_size')}|{getMetric('max_on_wire_size')}|\n"
+            with_info = True
+        else:
+            body_error += f"|{FAIL_CHAR}|{gate['name']}|{getMetric('current_on_disk_size')}|{getMetric('max_on_disk_size')}|{getMetric('current_on_wire_size')}|{getMetric('max_on_wire_size')}|\n"
+            body_error_footer += f"|{gate['name']}|{gate['error_type']}|{gate['message'].replace("\n", "<br>")}|\n"
+            with_error = True
+
+    body_error_footer += "\n</details>\n"
+    body = f"Please find below the results from static quality gates\n{body_error+body_error_footer if with_error else ""}\n\n{body_info if with_info else ""}"
```

Comment:
🥜 I'm surprised it didn't complain as you have double quotes inside double quotes. It's probably safer to change the internal ones to simple quotes, wdyt?
```suggestion
    body = f"Please find below the results from static quality gates\n{body_error+body_error_footer if with_error else ''}\n\n{body_info if with_info else ''}"
```

Reply:
True good catch 👍